addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,370 @@
+odoo.define('survey.form', function(require) {
+'use strict';
+
+var ajax = require('web.ajax');
+var field_utils = require('web.field_utils');
+var rpc = require('web.rpc');
+var publicWidget = require('web.public.widget');
+var time = require('web.time');
+
+publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
+    selector: '.o_survey_form',
+    events: {
+        'change .o_survey_form_simple_radio input[type="radio"]': '_onChangeRadio',
+        'change .o_survey_form_select': '_onChangeSelect',
+        'change .o_survey_form_checkbox_other': '_onChangeMultipleCheckbox',
+        'click button[name="button_submit"]': '_onSubmit',
+        'click .o_survey_header .breadcrumb-item a': '_onBreadcrumbClick',
+    },
+
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
+
+    /**
+    * @override
+    */
+    start: function () {
+        var self = this;
+        self.surveyToken = self.$target.data('surveyToken');
+        self.answerToken = self.$target.data('answerToken');
+
+        var $timer = $('.o_survey_timer');
+        if ($timer.length) {
+            self.surveyTimerWidget = new publicWidget.registry.SurveyTimerWidget(this);
+            self.surveyTimerWidget.attachTo($timer);
+            self.surveyTimerWidget.on('time_up', self, function (ev) {
+                self.$el.find('button[name="button_submit"]').click();
+            });
+        }
+
+        this.$('div.o_survey_form_date').each(function () {
+            self._updateDateForDisplay($(this));
```

Comment:
Some update_... code is used to automatically focus on the next comment box when choosing "other" in selection with other being a comment. I would make this more generic, like having a "o_survey_js_form_other_comment" in template, that automatically focus on next comment textarea. That way we don't have "radio / selection / checkboxes" specific code to run at each start, but some events defined in template and catched by JS.

Purpose behind that: trying to reduce code size with some generic elements and putting some stuff in template. Obviously we need some JS / custom stuff but it seems some code here could be cleaned further.